src/pip/_internal/network/session.py

```diff
@@ -281,8 +282,23 @@ def __init__(
         # Attach our User Agent to the request
         self.headers["User-Agent"] = user_agent()
 
+        no_prompt = MultiDomainBasicAuth(prompting=False, index_urls=index_urls)
+        prompt = MultiDomainBasicAuth(prompting=True, index_urls=index_urls)
+        prompt.passwords = no_prompt.passwords  # share same dict of passwords
+
         # Attach our Authentication handler to the session
-        self.auth = MultiDomainBasicAuth(index_urls=index_urls)
+        if enable_kerberos:
+            try:
+                from requests_kerberos import REQUIRED, HTTPKerberosAuth
```

Comment:
Do we have a benchmark of this import? If it is relatively quick, we should probably just always enable Kerberos whenever `requests_kerberos` is present in the environment, similar to how we handle `keyring` support.